faucet/vlan.py

```diff
@@ -56,13 +56,15 @@ class HostCacheEntry:
         'eth_src',
         'eth_src_int',
         'port',
+        'l3_src_ip',
     ]
 
-    def __init__(self, eth_src, port, cache_time):
+    def __init__(self, eth_src, port, cache_time, l3_src_ip=None):
```

Comment:
why does l3_src_ip need a default?

Reply:
trying to retain previous signature.

Reply:
In the big picture it's better to be precise rather than introduce variability. If it's not required then trying to retain something that doesn't matter is just increasing chaos.  Down with chaos.

(You should remove it)